### PR TITLE
tests speedup

### PR DESCRIPTION
**Pull request by naik-aakash**
- [x] Test adding adding splits tag to pytest for speedup

### CODE REVIEW

**Comment by naik-aakash** [comment]
Hi @JaGeo , This PR tries to resolve the open #97 issue. 

At the moment it seems to work with significant reduction in executing tests. Have attempted to use similar strategy of pymatgen. There seems to be one small issue, there seems to be some loss of coverage, mainly seems to stem from tests that used pytest-plugins. If you have any idea how to solve this let me know. I am looking into this as well. So far no luck 😞

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -27,7 +28,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install flake8 pytest pytest-mock
+        python -m pip install flake8 pytest pytest-mock pytest-split pytest-cov
         python -m pip install types-setuptools
         python -m pip install coverage
```
Comment (JaGeo) [inline .github/workflows/python-package.yml:33]: Do you need the second coverage when you install pytest-cov?

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -40,5 +41,30 @@ jobs:
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
     - name: Test with pytest and coverage
       run: |
-        coverage run -m pytest
-        coverage report
+        pytest --cov=lobsterpy --cov-append --splits 10 --group ${{ matrix.split }} --durations-path ./lobsterpy/.pytest-split-durations
```
Comment (JaGeo) [inline .github/workflows/python-package.yml:44]: To me it looks like you are doing 10 splits but only run 5 out of the 10? Do I overlook something?

**Comment by JaGeo** [comment]
I changed it already. Let's test the workflow with the 10 splits.

**Comment by JaGeo** [comment]
I will get back to this tomorrow.

**Review by JaGeo — APPROVED** [review]

**Comment by JaGeo** [comment]
I am merging it. Thanks @naik-aakash.

**Pull request merged**